### PR TITLE
Fix bug where delete menu appears for non-root posts

### PR DESCRIPTION
**Pull request by wiggin77**
#### Summary
This PR fixes a bug where the Delete Root post menu appears for posts that are not root posts.

I was relying on ReplyCount being accurate to determine if a post has replies. On the client the ReplyCount can become non-zero for non-root posts when more replies are added.  It gets fixed on page reload, which is why I missed it, but is inaccurate until refresh. 

The fix is to check that root_id is empty (meaning root post) and the reply count is non-zero.

#### Ticket Link
https://hub.mattermost.com/private-core/pl/stzhhdk3gtri7r8uezngo5xyww

### CODE REVIEW

File: webapp/src/index.tsx [hunk 1]
```diff
@@ -30,19 +30,21 @@ export default class Plugin {
...
-                // check if post is already marked as root post deleted
+                // hide menu if post is already marked as root post deleted
                 if (post.props && post.props.rootdel) {
                     return false;
                 }
 
-                // Check if the user has permissions to edit his own post or edit other's posts if not the author
+                // Check if the user has permissions to edit his own post, or other's posts if not the author.
+                // We check for edit permissions instead of delete since on the back-end we will be editing the
+                // post and making it immutable.
```
Comment (saturninoabril) [inline webapp/src/index.tsx:47]: Will there be a separate PR to add `delete` in addition to edit permissions?

**Review by saturninoabril — APPROVED** [review]
Thanks @wiggin77, tested and passed.

File: webapp/src/index.tsx (hunk 1 above)
Comment (wiggin77) [reply webapp/src/index.tsx:47]: I thought about it but can't find a reason to check delete permissions. Even removing attachments is considered a post edit in the other plugin.  The only permissions checked on the backend are for edit.

**Review by fmartingr — APPROVED** [review]
